api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -184,13 +184,13 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 			}
 			
 			//concept should be the same as on previous order, same applies to drug for drug orders
-			boolean isDrugOrderAndHasADrug = isDrugOrder && ((DrugOrder) order).getDrug() != null;
+			boolean isDrugOrderAndHasADrug = isDrugOrder && (((DrugOrder) order).getDrug() != null || ((DrugOrder) order).isNonCodedDrug()) ;
```

Comment:
DrugOrder.isNonCodedDrug() will never return true if getDrug order returns null if you implement isNonCodedDrug well where it checks if the drug's concept is equals to DRUG OTHER concept, so that check would be redundant, makes sense?